### PR TITLE
get-apps: rework search interaction, update UX

### LINKED ISSUE #166: landscape: rework software search interaction
Some changes to make software search easier to work with for new users:

- Remove search icon next to "Get Urbit apps"
- Move search field into modal and modify labels
- Provide link to UF app directory

[Figma with comments](https://www.figma.com/file/lPb6AIjQbg6ECbhxc6vXer/2023-Q2-Onboarding?type=design&node-id=419-26934&t=IwWbOfPUSNipBYkl-11)

Image
Image

### PR DESCRIPTION
**Pull request by patosullivan**
Closes #166

Updates react-router to v6. Fixes some issues with the search within system preferences as well.

Looks like this:

https://github.com/tloncorp/landscape/assets/1221094/83d28759-8031-41c5-971c-bfba2301b18a

Fixes LAND-511

### CODE REVIEW

**Comment by jamesacklin** [comment]
Works in all cases for me in Safari. I also tested:

- Opening the app preferences
- Searching through the preference panes
- Setting some security options
- Suspending and resuming an app
- Uninstalling and reinstalling an app

File: ui/src/nav/Nav.tsx [hunk 1]
```diff
@@ -219,32 +179,32 @@ export const Nav: FunctionComponent<NavProps> = ({ menu }) => {
       />
       <Dialog open={isOpen} onOpenChange={onDialogClose}>
         <DialogContent
-          onInteractOutside={preventClose}
-          onOpenAutoFocus={onOpen}
-          className="scroll-left-50 scroll-full-width outline-none fixed bottom-0 z-50 flex h-full max-h-full max-w-[882px] -translate-x-1/2 flex-col justify-end px-4 text-gray-400 sm:top-0 sm:bottom-auto sm:h-auto sm:justify-start sm:pb-4"
+          // onOpenAutoFocus={onOpen}
+          onInteractOutside={() => onDialogClose(false)}
```
Comment (arthyn) [inline ui/src/nav/Nav.tsx:183]:
```suggestion
```

**Review by arthyn — APPROVED** [review]
lgtm

File: ui/src/app.tsx [hunk 2]
```diff
@@ -24,6 +23,14 @@ import { useBrowserId, useLocalState } from './state/local';
...
 import { useErrorHandler } from './logic/useErrorHandler';
 import useSchedulerStore, { useScheduler } from './state/scheduler';
+import { TileInfo } from './tiles/TileInfo';
+import { SuspendApp } from './tiles/SuspendApp';
+import { RemoveApp } from './tiles/RemoveApp';
+import { Notifications } from './nav/notifications/Notifications';
+import { SystemPreferences } from './preferences/SystemPreferences';
+import { Help } from './nav/Help';
+import GetApps from './nav/GetApps';
+import { Search } from './nav/Search';
```
[the remark this answers is not in the document]
Comment (patosullivan) [reply ui/src/app.tsx:33]: yep, forgot to remove that

**Pull request merged**